src/geometry/manifold/derivation_bundle.lean

```diff
@@ -0,0 +1,152 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nicolò Cavalleri
+-/
+
+import geometry.manifold.algebra.smooth_functions
+import ring_theory.derivation
+
+/-!
+
+# Derivation bundle
+
+In this file we define the derivations at a point of a manifold on the algebra of smooth fuctions.
+Moreover we define the total bundle of derivations (although at the moment it has not been given a
+topology). Finally we define the differential of a function in terms of derivations.
+
+The content of this file is not meant to be regarded as an alternative definition to the current
```

Comment:
I think it would be useful to sketch out the mathematical construction up here and to mention explicitly that (at least in finite dimensions) there is a natural equivalence:
```lean
example [smooth_manifold_with_corners I M] (x : M) :
  tangent_space I x ≃ₗ[𝕜] point_derivation I x :=
sorry
```

Reply:
I would rather include this comment when I actually prove this in the case of the reals, i.e. when `𝕜 = ℝ`. I believe this will happen rather soon, since partitions of unity are very close.